### PR TITLE
Validate vaccination status for JSON

### LINKED ISSUE #208: [PE-D] stats command does not work as specified
Here I added some students on my own (5 is vaccinated, 1 is not vaccinated according to my add command input). The GUI shows correctly 5 vaccinated and 1 not vaccinated. However, when I run stats command (stats all, stats NUS, stats SCI), all the value returned is 0%.

This scenario is created because the app is unable to load its own data files and I tried to work on empty data.

I tried to restart the app and try again but seems like data that I put in in the last usage is still there and stats function still does not work.

I think stats function does not work with empty data set or update, I tried clearing the data and just add 1 student vaccinated and then call stats, the result is still 0%

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Eriksen2411/ped#23

### PR DESCRIPTION
**Pull request by SiTingST**
Ensure proper validation of vaccination status when users edit JSON file directly is in place. 
Ensure that the correct label is shown for vaccination status when JSON file is tampered with. 
Resolves #208

### CODE REVIEW

File: src/main/java/seedu/student/model/student/VaccinationStatus.java [hunk 1]
```diff
@@ -23,13 +23,12 @@
...
         checkArgument(isValidStatus(vaccinationStatus), MESSAGE_CONSTRAINTS);
-        if (vaccinationStatus.equals("vaccinated")) {
+        if (vaccinationStatus.toLowerCase().equals("vaccinated")) {
             status = VaccinationStatusAbbreviation.VACCINATED;
-            textUI = vaccinationStatus;
         } else {
             status = VaccinationStatusAbbreviation.UNVACCINATED;
-            textUI = vaccinationStatus;
         }
+        textUI = status.toString().toLowerCase();
```
Comment (dannn-tran) [inline src/main/java/seedu/student/model/student/VaccinationStatus.java:31]: is this field `textUI` used anywhere? it seems that no method uses it.

**Review by dannn-tran — COMMENTED** [review]
@SiTingST another small thing is that maybe the enum class can be named as `VaccinationStatus`, because they are not abbreviations?

File: src/main/java/seedu/student/model/student/VaccinationStatus.java (hunk 1 above)
Comment (SiTingST) [reply src/main/java/seedu/student/model/student/VaccinationStatus.java:31]: It's being used in the UI side (`StudentCard.java`) in which the vaccination status label will be set to `textUI`.

**Review by dannn-tran — APPROVED** [review]
LGTM

**Pull request merged**